src/components/QRCodeModal/index.tsx

```diff
@@ -0,0 +1,84 @@
+import QRCode from 'qrcode.react';
+import React, { useEffect, useRef } from 'react';
+import { FcDownload } from 'react-icons/fc';
+import { IoCloseSharp } from 'react-icons/io5';
+
+import Button from '@/components/Button';
+const RDSIcon = '_next/image?url=%2Frds.png&w=64&q=75';
+
+interface QRCodeModalProps {
+    value: string;
+    onClose: () => void;
+}
+
+const QRCodeModal: React.FC<QRCodeModalProps> = ({ value, onClose }) => {
+    const modalRef = useRef<HTMLDivElement>(null);
+
+    const handleClickOutside = (event: MouseEvent) => {
+        if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
+            onClose();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside);
+
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [onClose]);
+
+    const downloadQRCode = () => {
+        const canvas = document.getElementById('qr-code') as HTMLCanvasElement;
+        if (canvas) {
+            const pngUrl = canvas.toDataURL('image/png').replace('image/png', 'image/octet-stream');
+            const downloadLink = document.createElement('a');
+            downloadLink.href = pngUrl;
+            downloadLink.download = 'qr-code.png';
```

Comment:
UX improvement: downloaded image name can include original url so that each downloaded qr code image is distinguishable without scanning and checking.

Reply:
> UX improvement: downloaded image name can include original url so that each downloaded qr code image is distinguishable without scanning and checking.

original url may have special chars, and very log too like what if i want to generate for google sheet, can we have last 5-6 chars from short url

Reply:
Alternatively, we can designate the name "shortUrl" for the QR code.

Reply:
> Alternatively, we can designate the name "shortUrl" for the QR code.

what if i will download multiple QR code..?